### PR TITLE
Enable dataflow-beam-import integration tests

### LINKED ISSUE #2623: Enable dataflow-beam-import integration tests
There are two integration tests which never executed:

- `bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/sequencefiles/EndToEndIT.java`
- `bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/it/CloudBigtableBeamITTest.java`

They could be added to the regular execution to increase the coverage.

### PR DESCRIPTION
**Pull request by dmitry-fa**
Fixes #2623

### CODE REVIEW

**Comment by dmitry-fa** [comment]
I need the bucket name to run sequencefileIntegrationTest and beamIntegrationTest profiles

**Comment by kolea2** [comment]
@dmitry-fa I created the bucket 'java-bigtable-hbase-testing' to use for these tests (gsutil link: gs://java-bigtable-hbase-testing)

File: bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/sequencefiles/EndToEndIT.java [hunk 1]
```diff
@@ -51,47 +47,34 @@
...
-
-    dataflowStagingLocation = getTestProperty(GOOGLE_DATAFLOW_STAGING_LOCATION);
-
-    cloudTestDataFolder = getTestProperty(CLOUD_TEST_DATA_FOLDER);
-    if (!cloudTestDataFolder.endsWith(File.separator)) {
-      cloudTestDataFolder = cloudTestDataFolder + File.separator;
-    }
-
-    // Cloud Storage config
-    GcpOptions gcpOptions = PipelineOptionsFactory.create().as(GcpOptions.class);
```
Comment (kolea2) [inline bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/sequencefiles/EndToEndIT.java:88]: why was this removed?

**Review by kolea2 — CHANGES_REQUESTED** [review]

File: bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/it/CloudBigtableBeamITTest.java [hunk 2]
```diff
@@ -117,21 +120,24 @@
...
+      Integer.getInteger("google.dataflow.total_row_count", 1_000_000);
+  private static final int PREFIX_COUNT = Integer.getInteger("google.dataflow.prefix_count", 1_000);
 
   @BeforeClass
   public static void setUpConfiguration() {
-    Preconditions.checkArgument(stagingLocation != null, "Set -D" + STAGING_LOCATION_KEY + ".");
-    Preconditions.checkArgument(zoneId != null, "Set -D" + ZONE_ID_KEY + ".");
-    Preconditions.checkArgument(projectId != null, "Set -D" + PROJECT_ID_KEY + ".");
-    Preconditions.checkArgument(instanceId != null, "Set -D" + INSTANCE_ID_KEY + ".");
+    projectId = TestHelper.getTestProperty(PROJECT_ID_KEY);
```
[the remark this answers is not in the document]
Comment (dmitry-fa) [reply bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/it/CloudBigtableBeamITTest.java:135]: not, it's not removed, it's just unified to be used by all tests:

TestHelper.getTestProperty is implemented as:
```
  public static String getTestProperty(String name) {
    return checkNotNull(System.getProperty(name), "Required property missing: " + name);
  }
```

File: bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/sequencefiles/EndToEndIT.java (hunk 1 above)
Comment (dmitry-fa) [reply bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/sequencefiles/EndToEndIT.java:88]: It's not removed, it moved to TestHelper for the sake of unification.

TestHelper.initGcsUtil:
```
  private static GcsUtil initGcsUtil(String projectId) {
    GcpOptions gcpOptions = PipelineOptionsFactory.create().as(GcpOptions.class);
    gcpOptions.setProject(projectId);
    return new GcsUtil.GcsUtilFactory().create(gcpOptions);
  }
```

File: .kokoro/nightly/integration.cfg [hunk 3]
```diff
@@ -8,7 +8,7 @@ env_vars: {
 
 env_vars: {
     key: "INTEGRATION_TEST_ARGS"
-    value: "-P bigtableIntegrationTest,bigtableIntegrationTestH2 -Dgoogle.bigtable.project.id=gcloud-devel -Dgoogle.bigtable.instance.id=google-cloud-bigtable"
+    value: "-P bigtableIntegrationTest,bigtableIntegrationTestH2,sequencefileIntegrationTest,beamIntegrationTest -Dgoogle.dataflow.gcsPath=gs://java-bigtable-hbase-testing -Dgoogle.bigtable.project.id=gcloud-devel -Dgoogle.bigtable.instance.id=google-cloud-bigtable"
```
[the remark this answers is not in the document]
Comment (dmitry-fa) [reply .kokoro/nightly/integration.cfg]: Sure, I will add.

There is a problem with the third profile `bigtableDataflowIntegrationTest`
The test is unstable, it fails periodically with: 
```
[INFO] Running com.google.cloud.bigtable.beam.CloudBigtableIOIntegrationTest
2020-09-25 18:28:48 INFO  CloudBigtableIOIntegrationTest:90 - Creating table in testEstimatedAndSplitForLargeTable()
2020-09-25 18:28:52 INFO  CloudBigtableIOIntegrationTest:90 - Adding 1000 rows in testEstimatedAndSplitForLargeTable()
2020-09-25 18:29:03 ERROR AbstractRetryingOperation:158 - Could not complete RPC. Failure #0, got: Status{code=INTERNAL, description=Operation was interrupted, cause=null} on channel null.
Trailers: Metadata()
2020-09-25 18:29:03 WARN  BigtableBufferedMutator:130 - Exception occurred in BufferedMutator
com.google.bigtable.repackaged.io.grpc.StatusRuntimeException: INTERNAL: Operation was interrupted
	at com.google.bigtable.repackaged.io.grpc.Status.asRuntimeException(Status.java:524)
```
I will not include it for now and will investigate the problem.

File: bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/sequencefiles/EndToEndIT.java [hunk 4]
```diff
@@ -17,25 +17,21 @@
 
 import static com.google.common.base.Preconditions.checkNotNull;
 
+import com.google.cloud.bigtable.beam.TestHelper;
```
Comment (kolea2) [inline bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/sequencefiles/EndToEndIT.java:20]: From the logs for this test, I see:
```
2020-09-25 13:20:15 INFO  DataflowPipelineOptions$StagingLocationFactory:186 - No stagingLocation provided, falling back to gcpTempLocation
```

is this creating a temp bucket (and cleaning it up after)? Should we create another one?

Comment (kolea2) [reply bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/sequencefiles/EndToEndIT.java:20]: Additionally from the logs, `PackageUtil` creates a lot of logging statments. Can we turn down the logging level to reduce some of the noise?

Comment (kolea2) [reply bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/sequencefiles/EndToEndIT.java:20]: Finally, in the cleanup it looks like it's trying to delete a folder that already does not exist.

File: bigtable-dataflow-parent/bigtable-hbase-beam/src/test/java/com/google/cloud/bigtable/beam/CloudBigtableIOIntegrationTest.java [hunk 5]
```diff
@@ -18,6 +18,7 @@
 import com.google.bigtable.repackaged.com.google.cloud.bigtable.config.Logger;
 import com.google.bigtable.repackaged.com.google.cloud.bigtable.data.v2.models.KeyOffset;
 import com.google.cloud.bigtable.hbase.BigtableConfiguration;
+import com.google.cloud.bigtable.hbase.BigtableOptionsFactory;
```
Comment (kolea2) [inline bigtable-dataflow-parent/bigtable-hbase-beam/src/test/java/com/google/cloud/bigtable/beam/CloudBigtableIOIntegrationTest.java:21]: same question here if we can lessen the logging levels for these tests

File: bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/sequencefiles/EndToEndIT.java (hunk 4 above)
Comment (dmitry-fa) [reply bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/sequencefiles/EndToEndIT.java:20]: > From the logs for this test, I see:
> 
> ```
> 2020-09-25 13:20:15 INFO  DataflowPipelineOptions$StagingLocationFactory:186 - No stagingLocation provided, falling back to gcpTempLocation
> ```
> 
> is this creating a temp bucket (and cleaning it up after)? Should we create a bucket explicitly instead?

No, these tests never create a bucket. If 'stagingLocation' path is not provided, a subfolder will be used in `tempLocation`

Comment (dmitry-fa) [reply bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/sequencefiles/EndToEndIT.java:20]: > Additionally from the logs, `PackageUtil` creates a lot of logging statements. Can we turn down the logging level to reduce some of the noise?

As I can see tests are running with LoggingLevel=info, which IHMO produces reasonable amount of messages. `PackageUtil` is external class, therefore we can't update it to lessen the level of noising messages, we can only shut it up.
The messages from `PackageUtil` are not useless and they might become valueable for those who degug this test.
So, I would allow `PackageUtil` to talk.

Comment (dmitry-fa) [reply bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/sequencefiles/EndToEndIT.java:20]: > 
> 
> Finally, in the cleanup it looks like it's trying to delete a folder that already does not exist.

I fixed it. Not sure if you like the fix.

File: bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/TestHelper.java [hunk 6]
```diff
@@ -49,7 +50,13 @@ public static void cleanUpStorageFolder(String gsPath) throws IOException {
     for (GcsPath path : gcsUtil.expand(GcsPath.fromUri(gsPath + "*"))) {
       pathsToRemove.add(path.toString());
     }
-    pathsToRemove.add(gsPath);
+    try {
+      if (gcsUtil.getObject(GcsPath.fromUri(gsPath)) != null) {
```
Comment (kolea2) [inline bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/TestHelper.java:54]: how can the path exist?

Comment (dmitry-fa) [reply bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/TestHelper.java:54]: This path is probably created when Import/Export tools are working, not by the tests.
The tests do something like:

```
  BeforeClass:
    gcsPath = TestHelper.getTestProperty("google.dataflow.gcsPath");
    gcsWorkDir = gcsPath + "/" + UUID.randomUUID().toString();
    stagingLocation = gcsWorkDir + "/staging";
    tempLocation = gcsWorkDir + "/temp";


  During testing:
    DataflowPipelineOptions options = PipelineOptionsFactory.as(DataflowPipelineOptions.class);
    options.setStagingLocation(stagingLocation);
    options.setTempLocation(tempLocation);


  AfterClass:
    TestHelper.cleanUpStorageFolder(stagingLocation);
    TestHelper.cleanUpStorageFolder(tempLocation);
    TestHelper.cleanUpStorageFolder(gcsWorkDir);
```

Without removing the path under question the bucket is not properly cleaned up:
```
#> gsutil ls gs://bucket_for_bt_test/
gs://bucket_for_bt_test/546d262a-8a5b-41c9-b0ad-e12415576238/
#> gsutil ls gs://bucket_for_bt_test/546d262a-8a5b-41c9-b0ad-e12415576238/
gs://bucket_for_bt_test/546d262a-8a5b-41c9-b0ad-e12415576238/temp/
```

Ideally, it should be enough to call `TestHelper.cleanUpStorageFolder(gcsWorkDir)`, but I didn't find how to perform
`gsutil rm -rf` with currently used `org.apache.beam.sdk.extensions.gcp.util.GcsUtil`.
So, removing the extra path helps to keep the bucket clear from the garbage.

**Review by kolea2 — COMMENTED** [review]
This looks good, will ask @igorbernstein2 to take a pass through as well

File: bigtable-dataflow-parent/bigtable-beam-import/pom.xml [hunk 7]
```diff
@@ -300,26 +300,24 @@ limitations under the License.
...
-         google.dataflow.stagingLocation
-         cloud.test.data.folder
+         google.dataflow.gcsPath
      -->
       <build>
         <plugins>
           <plugin>
             <groupId>org.apache.maven.plugins</groupId>
-            <artifactId>maven-failsafe-plugin</artifactId>
+            <artifactId>maven-surefire-plugin</artifactId>
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-beam-import/pom.xml:309]: why surefire? shouldnt we use failsafe for integration tests?

File: bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/it/CloudBigtableBeamITTest.java [hunk 8]
```diff
@@ -117,21 +120,24 @@
   private static final String adminEndpoint =
       System.getProperty(BIGTABLE_ADMIN_HOST_KEY, BIGTABLE_ADMIN_HOST_DEFAULT);
   private static final String TABLE_NAME_STR =
-      System.getProperty("tableName", "BeamCloudBigtableIOIntegrationTest");
+      System.getProperty("google.dataflow.tableName", "BeamCloudBigtableIOIntegrationTest");
 
   private static final TableName TABLE_NAME = TableName.valueOf(TABLE_NAME_STR);
-  private static final byte[] FAMILY = Bytes.toBytes("test-family");
+  private static final byte[] FAMILY = Bytes.toBytes("google.dataflow.test-family");
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/it/CloudBigtableBeamITTest.java:126]: why does the table schema need to be configurable?

File: bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/it/CloudBigtableBeamITTest.java [hunk 9]
```diff
@@ -117,21 +120,24 @@
...
   private static final byte[] QUALIFIER = Bytes.toBytes("test-qualifier");
-  private static final int CELL_SIZE = Integer.getInteger("cell_size", 1_000);
-  private static final long TOTAL_ROW_COUNT = Integer.getInteger("total_row_count", 1_000_000);
-  private static final int PREFIX_COUNT = Integer.getInteger("prefix_count", 1_000);
+  private static final int CELL_SIZE = Integer.getInteger("google.dataflow.cell_size", 1_000);
+  private static final long TOTAL_ROW_COUNT =
+      Integer.getInteger("google.dataflow.total_row_count", 1_000_000);
+  private static final int PREFIX_COUNT = Integer.getInteger("google.dataflow.prefix_count", 1_000);
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/it/CloudBigtableBeamITTest.java:131]: I believe we have some internal jobs that depend on the existing prop names. I think namespacing them is a good idea, but I think it needs to happen in 2 steps:
1. if the new name isnt present fallback to the old name before falling back to a default
2. once all jobs have been migrate, drop the fallback

Also, I think think it should be something like google.bigtable.dataflow to avoid triggering something unintended

File: bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/sequencefiles/EndToEndIT.java [hunk 10]
```diff
@@ -51,47 +47,34 @@
...
-
   // Column family name used in all test bigtables.
   private static final String CF = "column_family";
 
-  // Full path of the Cloud Storage folder where dataflow jars are uploaded to.
-  private static final String GOOGLE_DATAFLOW_STAGING_LOCATION = "google.dataflow.stagingLocation";
+  private static final String projectId = getTestProperty(BigtableOptionsFactory.PROJECT_ID_KEY);
+  private static final String instanceId = getTestProperty(BigtableOptionsFactory.INSTANCE_ID_KEY);
+  private static final String gcsPath = getTestProperty("google.dataflow.gcsPath");
+  private static final String gcsWorkDir = gcsPath + "/" + UUID.randomUUID();
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/sequencefiles/EndToEndIT.java:69]: similar to above

**Review by igorbernstein2 — CHANGES_REQUESTED** [review]

File: bigtable-dataflow-parent/bigtable-beam-import/pom.xml (hunk 7 above)
Comment (dmitry-fa) [reply bigtable-dataflow-parent/bigtable-beam-import/pom.xml:309]: `surefire` to be similar to other integration tests:
- https://github.com/googleapis/java-bigtable-hbase/blob/master/bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml#L42-L51
- https://github.com/googleapis/java-bigtable-hbase/blob/master/bigtable-hbase-2.x-parent/bigtable-hbase-2.x-integration-tests/pom.xml#L50-L56
- https://github.com/googleapis/java-bigtable-hbase/blob/master/bigtable-dataflow-parent/bigtable-beam-import/pom.xml#L336-L342

there must be some reasons why `surefire` is used.

File: bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/it/CloudBigtableBeamITTest.java (hunk 8 above)
Comment (dmitry-fa) [reply bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/it/CloudBigtableBeamITTest.java:126]: It's still hardcoded, changed by mistake. I will restore the original one value.

File: bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/sequencefiles/EndToEndIT.java (hunk 10 above)
Comment (dmitry-fa) [reply bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/sequencefiles/EndToEndIT.java:69]: `google.dataflow.gcsPath` is introduced by this fix, fallback is not required

File: bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/it/CloudBigtableBeamITTest.java (hunk 9 above)
Comment (dmitry-fa) [reply bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/it/CloudBigtableBeamITTest.java]: done, please take a look

**Comment by igorbernstein2** [comment]
This looks like it was done in a separate PR
